azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -15,34 +15,405 @@
+      logger.info("Kube config path is : {}", kubeConfigPath);
+      this.client =
+          ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(
+              Files.newBufferedReader(Paths.get(kubeConfigPath), Charset.defaultCharset())))
+              .build();
+      this.coreV1Api = new CoreV1Api(client);
+    } catch (IOException exception) {
+      logger.error("Unable to read kube config file: {}", exception.getMessage());
+      throw new ExecutorManagerException(exception);
+    }
   }
 
+  /**
+   * This method is used to create container during dispatch of execution. It will create pod for a
+   * flow execution. It will also create a service for a pod if azkaban.kubernetes.service .required
+   * property is set.
+   *
+   * @param executionId
+   * @throws ExecutorManagerException
+   */
   @Override
   public void createContainer(final int executionId) throws ExecutorManagerException {
-    logger.info("Container is created");
+    createPod(executionId);
+    if (isServiceRequired()) {
+      createService(executionId);
+    }
   }
 
+  /**
+   * This method is used to delete container. It will delete pod for a flow execution. If the
+   * service was created then it will also delete the service. This method can be called as a part
+   * of cleanup process for containers in case containers didn't shutdown gracefully.
+   *
+   * @param executionId
+   * @throws ExecutorManagerException
+   */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    logger.info("Container is deleted");
+    deletePod(executionId);
+    if (isServiceRequired()) {
+      deleteService(executionId);
+    }
+  }
+
+  /**
+   * This method is used to create pod. 1. Fetch jobTypes for the flow 2. Fetch flow parameters for
+   * version set and each image type if it is set. 3. If valid version set is provided then use
+   * versions from it. 4. If valid version set is not provided then call Ramp up manager API and get
+   * image version for each image type. 5. Add all the validation around a) whether version set is
+   * valid or not. b) If it is valid then is there any change in flow and new jobType is introduced
+   * after version set was created? If so, create new version set using versions mentioned in
+   * version set and ramp up for new jobType. 6. Create pod spec using all the version information
+   * 7. Insert version set into execution_flows tables for a reference 8. Emit version set as a part
+   * of flow life cycle event.
+   *
+   * @param executionId
+   * @throws ExecutorManagerException
+   */
+  private void createPod(final int executionId) throws ExecutorManagerException {
+    // Fetch execution flow from execution Id.
+    final ExecutableFlow flow = executorLoader.fetchExecutableFlow(executionId);
+    // Step 1: Fetch set of jobTypes for a flow from executionId
+    TreeSet<String> jobTypes = getJobTypesForFlow(flow);
```

Comment:
nit: final, applies to all the declarations below too.

Reply:
Done